wallet/src/repository/wallet/mod.rs

```diff
@@ -1336,25 +1336,39 @@ where
         }
 
         let mut output_amount: u64 = 0;
+        let mut own_outputs: HashMap<PublicKeyHash, model::OutputType> = HashMap::new();
         for (index, output) in outputs.iter().enumerate() {
-            if self
-                .db
-                .get_opt::<_, model::Path>(&keys::pkh(&output.pkh))?
+            if let Some(path) = self.db.get_opt::<_, model::Path>(&keys::pkh(&output.pkh))? {
+                match path.keychain {
+                    x if x == constants::EXTERNAL_KEYCHAIN => {
+                        own_outputs.insert(output.pkh, model::OutputType::External);
+                    }
+                    x if x == constants::INTERNAL_KEYCHAIN => {
+                        own_outputs.insert(output.pkh, model::OutputType::Internal);
+                    }
+                    _ => {
+                        log::warn!("Output found in DB but keychain is not known");
```

Comment:
Log some more information: 

```suggestion
                        log::warn!("Output found in DB but keychain is not known: {}", output.pkh);
```